homeassistant/components/systemair/climate.py

```diff
@@ -0,0 +1,224 @@
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.climate import PLATFORM_SCHEMA, ClimateDevice
+from homeassistant.components.climate.const import (
+    ATTR_FAN_MODE, ATTR_HVAC_MODE, ATTR_PRESET_MODE, FAN_HIGH, FAN_LOW,
+    FAN_MEDIUM, FAN_OFF, HVAC_MODE_AUTO, HVAC_MODE_OFF, SUPPORT_FAN_MODE,
+    SUPPORT_PRESET_MODE, SUPPORT_TARGET_TEMPERATURE)
+from homeassistant.const import ATTR_TEMPERATURE, CONF_NAME, TEMP_CELSIUS
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+from homeassistant.components.systemair import (
+    ATTR_CURRENT_HUMIDITY, ATTR_TARGET_TEMPERATURE,
+    PRESET_AUTO, PRESET_CROWDED, PRESET_FIREPLACE, PRESET_HOLIDAY, PRESET_IDLE,
+    PRESET_MANUAL, PRESET_REFRESH, SIGNAL_SYSTEMAIR_UPDATE_DONE)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_NAME): cv.string,
+})
+
+_LOGGER = logging.getLogger(__name__)
+
+HA_STATE_TO_SYSTEMAIR = {
+    HVAC_MODE_AUTO: 'auto',
+    HVAC_MODE_OFF: 'off',
+
+}
+
+HA_PRESET_TO_SYSTEMAIR = {
+    PRESET_AUTO: 'auto',
+    PRESET_MANUAL: 'manual',
+    PRESET_CROWDED: 'crowded',
+    PRESET_REFRESH: 'refresh',
+    PRESET_FIREPLACE: 'fireplace',
+    PRESET_HOLIDAY: 'holiday',
+    PRESET_IDLE: 'idle'
+
+}
+
+FAN_MAXIMUM = 'maximum'
+
+HA_SET_ATTR_TO_SYSTEMAIR = {}
+HA_ATTR_TO_SYSTEMAIR = {}
+
+PRESET_MODE = {}
+FAN_MODE = {}
+
+SYSTEMAIR_TRANSLATE_TO_HA = {
+    ATTR_FAN_MODE: FAN_MODE,
+    ATTR_PRESET_MODE: PRESET_MODE
+}
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Set up Systemair climate based on config_entry."""
+    from systemair.savecair.const import (
+        SENSOR_CURRENT_FAN_MODE,
+        SENSOR_TARGET_TEMPERATURE,
+        SENSOR_CURRENT_OPERATION,
+        SENSOR_TEMPERATURE_EXTRACT,
+        SENSOR_CUSTOM_OPERATION,
+        SENSOR_CURRENT_HUMIDITY,
+        SENSOR_MODE_CHANGE_REQUEST,
+        SENSOR_CUSTOM_FAN_MODE
+    )
+
+    global HA_ATTR_TO_SYSTEMAIR
+    HA_ATTR_TO_SYSTEMAIR = {
+        ATTR_TARGET_TEMPERATURE: SENSOR_TARGET_TEMPERATURE,
+        ATTR_FAN_MODE: SENSOR_CUSTOM_FAN_MODE,
+        ATTR_PRESET_MODE: SENSOR_CURRENT_OPERATION,
+        ATTR_TEMPERATURE: SENSOR_TEMPERATURE_EXTRACT,
+        ATTR_HVAC_MODE: SENSOR_CUSTOM_OPERATION,
+        ATTR_CURRENT_HUMIDITY: SENSOR_CURRENT_HUMIDITY,
+    }
+
+    global HA_SET_ATTR_TO_SYSTEMAIR
+    HA_SET_ATTR_TO_SYSTEMAIR = {
+        ATTR_TEMPERATURE: SENSOR_TARGET_TEMPERATURE,
+        ATTR_PRESET_MODE: SENSOR_MODE_CHANGE_REQUEST,
+        ATTR_FAN_MODE: SENSOR_CURRENT_FAN_MODE,
+        ATTR_HVAC_MODE: SENSOR_CUSTOM_OPERATION
+    }
+
+    sab = hass.data[config["platform"]]
+    async_add_entities([
+        SystemAIRClimate(hass, sab)
+    ])
+
+
+class SystemAIRClimate(ClimateDevice):
+    """Representation of a SystemAIR HVAC."""
+
+    def __init__(self, hass, sab):
```

Comment:
Don't pass in `hass`. It will be set on the entity when the entity has been added to home assistant. This means you can access `hass` with `self.hass`, after the entity has been added to home assistant.